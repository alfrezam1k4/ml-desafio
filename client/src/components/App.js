import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';

import { Layout } from '.';

const App = () => {
  return (
    <Router>
      <Route render={() => <Layout />} />
    </Router>
  );
};

export default App;
