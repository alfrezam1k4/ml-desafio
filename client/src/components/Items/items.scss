@import '../../utils/variables.scss';

.item-list {
  margin: 16px 32px;
}
.item-link {
  color: unset;
  text-decoration: unset;
}
.item-list:first-child {
  padding-top: 16px;
}
.item-list:not(:last-child) {
  border-bottom: 1pt solid #eee;
}
.item-image {
  width: 50px;
  border-radius: 4px;
  @media (min-width: $xs) {
    width: 100px;
  }
  @media (min-width: $sm) {
    width: 140px;
  }
  @media (min-width: $md) {
    width: 180px;
  }
}
ul {
  list-style-type: none;
  margin-block-start: -16px;
  margin-block-end: 0px;
  padding-inline-start: 0px;
  background-color: white;
}
.item-details {
  background-color: white;
  padding: 32px;
}
.content-image {
  display: flex;
  align-items: center;
  overflow: hidden;
}

.content-image img {
  max-width: 100%;
  display: block;
  height: auto;
}
.button-comprar {
  font-size: 16px;
  padding: 0 24px;
  height: 48px;
  border-radius: 6px;
  border-color: transparent;
  background-color: $button-blue;
  color: #fff;
  width: 100%;
  cursor: pointer;
}
.button-comprar:focus {
  outline: 0;
  box-shadow: 0 0 0 0.1875em rgba(30, 109, 255, 0.3);
  transition: box-shadow 0.25s ease-in;
}
.button-comprar:active {
  border-color: transparent;
  background-color: #1f4e96;
  color: #fff;
}
