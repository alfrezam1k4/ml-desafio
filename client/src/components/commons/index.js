import { Row, Col } from './Grid';
import Breadcrumb from './Breadcrumb';
export { Row, Col, Breadcrumb };
