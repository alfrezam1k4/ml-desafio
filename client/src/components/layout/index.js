import AppBar from './AppBar';
import Layout from './Layout';
import Background from './Background';

export { AppBar, Layout, Background };
