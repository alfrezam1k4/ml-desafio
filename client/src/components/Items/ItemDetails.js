import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';

import { withApi, formats } from '../../utils';
import { Breadcrumb, Col, Row } from '../commons';

const ItemDetails = ({ api, match }) => {
  const [product, setProduct] = useState(null);

  const [categories, setCategories] = useState([]);
  const [condition, setCondition] = useState('');
  useEffect(() => {
    api
      .getItemDetails(match.params.id)
      .then((res) => {
        if (res.data) {
          setProduct(res.data.product);
          switch (res.data.product.condition) {
            case 'new':
              setCondition('Nuevo');
              break;
            case 'used':
              setCondition('Usado');
              break;
            case 'refurbished':
              setCondition('Reacondicionado');
              break;

            default:
              break;
          }
          setCategories(res.data.categories);
        }
      })
      .catch((error) => {
        console.log(error);
      });
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <>
      <Breadcrumb categories={categories} />

      {product && (
        <div className="item-details">
          <div>
            <Row gutter={32}>
              <Col xs={24} md={16}>
                <div className="content-image">
                  <img src={product.picture} alt="" />
                </div>
              </Col>
              <Col xs={24} md={8}>
                <p
                  style={{ fontSize: '14px', marginBottom: '16px' }}
                >{`${condition} - ${product.sold_quantity} vendidos`}</p>
                <p style={{ fontSize: '24px', margin: 0, fontWeight: '600' }}>
                  {product.item.title}
                </p>
                <p
                  style={{
                    fontSize: '46px',
                    margin: '32px 0px',
                  }}
                >
                  {formats.moneyFormat(product.item.price.amount)}
                </p>
                <button className="button-comprar">Comprar</button>
              </Col>
            </Row>
            <Row>
              <Col xs={24} md={16}>
                <p
                  style={{
                    fontSize: '28px',
                    fontWeight: '600',
                    marginBottom: 0,
                  }}
                >
                  Descripcion del producto
                </p>
                <p style={{ fontSize: '16px', margin: '32px 0px' }}>
                  {product.description}
                </p>
              </Col>
            </Row>
          </div>
        </div>
      )}
    </>
  );
};

ItemDetails.propTypes = {};

export default withApi(ItemDetails);
