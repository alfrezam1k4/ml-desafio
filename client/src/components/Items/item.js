import React from 'react';
import PropTypes from 'prop-types';
import { Col, Row } from '../commons';
import { useResponsive, formats } from '../../utils';
import { icShipping } from '../../assets/icon';
import { Link } from 'react-router-dom';

const Item = ({ product }) => {
  const { price, title, address, picture, free_shipping } = product;
  const { screenIsUp } = useResponsive();

  return (
    <Link to={`/items/${product.id}`} className="item-link">
      <Row
        type="flex"
        justify="center"
        gutter={8}
        style={{ marginBottom: '16px' }}
      >
        <Col xs={24} sm={8} lg={5}>
          <img src={picture} alt="" className="item-image" />
        </Col>
        <Col
          xs={24}
          sm={12}
          lg={16}
          style={{ marginTop: screenIsUp('sm') ? '16px' : '0' }}
        >
          <>
            <p
              style={{
                fontSize: '24px',
                display: 'inline-block',
                margin: '16px 16px 32px 0px',
              }}
            >
              {formats.moneyFormat(price.amount)}
            </p>
            {free_shipping && (
              <img
                src={icShipping}
                alt="Envio"
                style={{ width: '24px', verticalAlign: 'text-bottom' }}
              />
            )}
          </>
          <p style={{ fontSize: '18px', maxWidth: '400px', marginTop: 0 }}>
            {title}
          </p>
        </Col>
        <Col
          xs={24}
          sm={4}
          lg={3}
          style={{ marginTop: screenIsUp('sm') ? '32px' : '0' }}
        >
          <p style={{ fontSize: '12px' }}>{address}</p>
        </Col>
      </Row>
    </Link>
  );
};

Item.propTypes = {
  product: PropTypes.object,
};

export default Item;
