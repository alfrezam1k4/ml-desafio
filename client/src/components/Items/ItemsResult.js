import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { withApi } from '../../utils';

import queryString from 'query-string';
import Item from './Item';
import './Items.scss';
import { Breadcrumb } from '../commons';

const ItemsResult = ({ api }) => {
  const { search } = queryString.parse(
    window.location.search + window.location.hash
  );
  const [products, setProducts] = useState([]);
  const [categories, setCategories] = useState([]);
  useEffect(() => {
    api
      .getItems(search)
      .then((res) => {
        if (res.data) {
          setProducts(res.data.items);
          setCategories(res.data.categories);
        }
      })
      .catch((error) => {
        console.log(error);
      });
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  return (
    <div>
      {products && products.length > 0 && (
        <>
          <Breadcrumb categories={categories} />
          <ul>
            {products.map((p) => (
              <li className="item-list" key={p.id}>
                <Item product={p} />
              </li>
            ))}
          </ul>
        </>
      )}
    </div>
  );
};

ItemsResult.propTypes = {
  api: PropTypes.object,
};

export default withApi(ItemsResult);
