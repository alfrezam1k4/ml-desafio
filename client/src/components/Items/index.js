import ItemsResult from './ItemsResult';
import Item from './Item';
import ItemDetails from './ItemDetails';

export { ItemsResult, Item, ItemDetails };
