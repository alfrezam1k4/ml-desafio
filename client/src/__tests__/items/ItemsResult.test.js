import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { ItemsResult } from '../../components/Items';

describe('<ItemsResult />', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <MemoryRouter>
        <ItemsResult />
      </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
