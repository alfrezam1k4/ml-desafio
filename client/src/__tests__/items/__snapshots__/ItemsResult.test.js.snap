// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemsResult /> renders correctly 1`] = `<div />`;
